File: problemSolving/BomberMan.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class BomberManResult {

    /*
     * Complete the 'bomberMan' function below.
     *
     * The function is expected to return a STRING_ARRAY.
     * The function accepts following parameters:
     * 1. INTEGER n
     * 2. STRING_ARRAY grid
     */

    public static List<String> bomberMan(int n, List<String> grid) {
        // Write your code here
        for (String string : grid) {
            string.replaceAll("O", "2");
        }

        for (int i = 1; i < n; i++) {
            if (i %2 == 1){

            }
            else{

            }

        }

    //     for (int i = 0; i < n; i++) {
    //         if (n % 3 == 1) {
    //             for (String row : grid) {
    //                 row.replaceAll("R", "O");
    //             }
    //         }
    //         if (n % 3 == 2) {
    //             for (String row : grid) {
    //                 row.replaceAll(".", "R");
    //             }
    //         }
    //         if (n % 3 == 0) {
    //             for (int row = 0; row < grid.size(); row++) {
    //                 for (int col = 0; col < grid.get(0).length(); col++) {
    //                     if (grid.get(row).charAt(col) == 'O') {
    //                         // explosion time (:
    //                         if (row + 1 <= grid.size()) {// bottom row

    //                             if (col + 1 <= grid.get(0).length()) {// right col
    //                                 String afterExplosionUp = grid.get(row - 1).substring(0, col) + '.'
    //                                         + grid.get(row - 1).substring(col + 1);
    //                                 grid.set(row - 1, afteafterExplosionUpExplosion);

    //                                 String afterExplosionLeft = grid.get(row).substring(0, col) + '.'
    //                                         + grid.get(row).substring(col - 1);
    //                                 grid.set(row, afterExplosionLeft);

    //                             }
    //                             if (col - 1 >= 0) {// left col
                                    
    //                             }

    //                         }
    //                         if (row - 1 >= 0) {// top row
                                
    //                         }
    //                         if (col + 1 <= grid.get(0).length()) {// right col
                                
    //                         }
    //                         if (col - 1 >= 0) {// left col
                                
    //                         }
    //                     }
    //                 }
    //             }
    //         }
    //     }
    // }
    }

public class BomberMan {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        // BufferedWriter bufferedWriter = new BufferedWriter(new
        // FileWriter(System.getenv("OUTPUT_PATH")));

        String[] firstMultipleInput = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        int r = Integer.parseInt(firstMultipleInput[0]);

        int c = Integer.parseInt(firstMultipleInput[1]);

        int n = Integer.parseInt(firstMultipleInput[2]);

        List<String> grid = IntStream.range(0, r).mapToObj(i -> {
            try {
                return bufferedReader.readLine();
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        })
                .collect(toList());

        List<String> result = BomberManResult.bomberMan(n, grid);
        System.out.println(result);
        // bufferedWriter.write(
        // result.stream()
        // .collect(joining("\n"))
        // + "\n"
        // );

        bufferedReader.close();
        // bufferedWriter.close();
    }
}

File: python/oneMonthPrepKit/bomberMan.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'bomberMan' function below.
#
# The function is expected to return a STRING_ARRAY.
# The function accepts following parameters:
#  1. INTEGER n
#  2. STRING_ARRAY grid
#

def bomberMan(n, grid):
    # Write your code here
    if n == 1:
        return grid
    fullbombs = ['O' * len(grid[0]) for i in range(len(grid))]
    if n%2 == 0:
        return fullbombs

    state2 = [row[:] for row in fullbombs]
    for i in range(len(grid)):
        for j in range(len(grid[0])):
            if grid[i][j] == 'O':
                if i - 1 >=0:
                    state2[i-1] = state2[i-1][:j] + '.' + state2[i-1][j+1:]
                if i + 1 <= len(grid)-1:
                    state2[i+1] = state2[i+1][:j] + '.' + state2[i+1][j+1:]
                if j + 1 <= len(grid[0])-1:
                    state2[i] = state2[i][:j+1] + '.' + state2[i][j+2:]
                if j - 1 >= 0:
                    state2[i] = state2[i][:j-1] + '.' + state2[i][j:]
                state2[i] = state2[i][:j] + '.' + state2[i][j+1:]

    state3 = [row[:] for row in fullbombs]
    for i in range(len(grid)):
        for j in range(len(grid[0])):
            if state2[i][j] == 'O':
                if i - 1 >=0:
                    state3[i-1] = state3[i-1][:j] + '.' + state3[i-1][j+1:]
                if i + 1 <= len(grid)-1:
                    state3[i+1] = state3[i+1][:j] + '.' + state3[i+1][j+1:]
                if j + 1 <= len(grid[0])-1:
                    state3[i] = state3[i][:j+1] + '.' + state3[i][j+2:]
                if j - 1 >= 0:
                    state3[i] = state3[i][:j-1] + '.' + state3[i][j:]
                state3[i] = state3[i][:j] + '.' + state3[i][j+1:]

    if n % 4 == 1:
        return state3

    if n %4 == 3:
        return state2

if __name__ == '__main__':
    fptr = sys.stdout

    first_multiple_input = input().rstrip().split()

    r = int(first_multiple_input[0])

    c = int(first_multiple_input[1])

    n = int(first_multiple_input[2])

    grid = []

    for _ in range(r):
        grid_item = input()
        grid.append(grid_item)

    result = bomberMan(n, grid)

    fptr.write('\n'.join(result))
    fptr.write('\n')

    fptr.close()
